pkg/operator/ceph/csi/spec.go

```diff
@@ -213,12 +217,23 @@ func StartCSIDrivers(namespace string, clientset kubernetes.Interface, ver *vers
 			return fmt.Errorf("failed to start rbdplugin daemonset: %+v\n%+v", err, rbdPlugin)
 		}
 	}
+
+	tolerations := getToleration()
+	nodeAffinity := getNodeAffinity()
 	if rbdProvisionerSTS != nil {
+		rbdProvisionerSTS.Spec.Template.Spec.Tolerations = tolerations
+		rbdProvisionerSTS.Spec.Template.Spec.Affinity = &corev1.Affinity{
+			NodeAffinity: nodeAffinity,
+		}
 		err = k8sutil.CreateStatefulSet("csi-rbdplugin-provisioner", namespace, clientset, rbdProvisionerSTS)
 		if err != nil {
 			return fmt.Errorf("failed to start rbd provisioner statefulset: %+v\n%+v", err, rbdProvisionerSTS)
 		}
 	} else if rbdProvisionerDeployment != nil {
+		rbdProvisionerDeployment.Spec.Template.Spec.Tolerations = tolerations
```

Comment:
I'd suggest a helper method to apply the tolerations and affinity so it isn't duplicated. For example, search for the method `ApplyToPodSpec()` for this pattern.

Reply:
sure